File: templates/Wpf/Pages/Settings.CodeBehind/Converters/EnumToBooleanConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Param_RootNamespace.Converters
{
    public class EnumToBooleanConverter : IValueConverter
    {
        public Type EnumType { get; set; }

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (parameter is string enumString)
            {
                if (Enum.IsDefined(EnumType, value))
                {
                    var enumValue = Enum.Parse(EnumType, enumString);

                    return enumValue.Equals(value);
                }
            }

            return false;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (parameter is string enumString)
            {
                return Enum.Parse(EnumType, enumString);
            }

            return null;
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Uwp/Pages/Settings.CodeBehind._VB/Helpers/EnumToBooleanConverter.vb
Namespace Helpers
    Public Class EnumToBooleanConverter
        Implements IValueConverter

        Public Property EnumType As Type

        Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
            If TypeOf parameter Is String Then
                If Not [Enum].IsDefined(EnumType, value) Then
                    Throw New ArgumentException("value must be an Enum!")
                End If

                Dim enumValue = [Enum].Parse(EnumType, parameter.ToString())

                Return enumValue.Equals(value)
            End If

            Throw New ArgumentException("parameter must be an Enum name!")
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
            Dim enumString = TryCast(parameter, String)
            If enumString IsNot Nothing Then
                Return [Enum].Parse(EnumType, enumString)
            End If

            Throw New ArgumentException("parameter must be an Enum name!")
        End Function
    End Class
End Namespace
